insert.go

```diff
@@ -190,3 +190,18 @@ func (b InsertBuilder) Values(values ...interface{}) InsertBuilder {
 func (b InsertBuilder) Suffix(sql string, args ...interface{}) InsertBuilder {
 	return builder.Append(b, "Suffixes", Expr(sql, args...)).(InsertBuilder)
 }
+
+// SetMap set columns and values for insert builder from a map of column name and value
+// note that it will reset all previous columns and values was set if any
+func (b InsertBuilder) SetMap(clauses map[string]interface{}) InsertBuilder {
+	cols := make([]string, len(clauses))
+	vals := make([]interface{}, len(clauses))
+	idx := 0
+	for cols[idx], vals[idx] = range clauses {
+		idx++
+	}
```

Comment:
I think this would be slightly more idiomatic:

```go
cols := make([]string, 0, len(clauses))
vals := make([]interface{}, 0, len(clauses))
for col, val := range clauses {
	cols = append(cols, col)
	vals = append(vals, val)
}
```